### ISSUE TITLE
Detect sound

### ISSUE THREAD

**Issue by nigeon** [puppeteer/puppeteer#516]
I would like to detect if a page is playing some sound. Is there a way to do that?

**Comment by ebidel** [comment]
This is a great question for stackoverflow :) Mind asking over there and posting back the link?

**Comment by aslushnikov** [comment]
What's your usecase?

**Comment by nigeon** [comment]
There's already Stackoverflow questions about this....:
- [Chrome tabs / Only html5](https://stackoverflow.com/questions/41649874/how-to-detect-if-a-chrome-tab-is-playing-audio)
- [Chrome tabs](https://stackoverflow.com/questions/40356873/is-it-possible-to-detect-whether-a-browser-tab-is-playing-audio-or-not)

I was referring is this was possible via Puppeteer and/or Chrome DevTools, as I unsuccessfully already searched for it. If is not, you can think it as a "feature request".

I'm building an ad scanner platform and need to detect if any sound is played. This adverts can be html5, flash, silverlight... basically anything that can be in a browser, and I need to detect if any of this is playing audio.

Chrome has this detection built-in in the tab (?) level (speaker icon). As the previous links state, there's an API doing what I need (https://developer.chrome.com/extensions/tabs) but for Chrome extensions and I think it would be great if this property is accessible via Puppeteer.

Thanks.

**Comment by ebidel** [comment]
Yep, as you point out, the plumbing is already there in Chrome to know if the tab is playing audio (it adds the speaker icon). The extensions API solves this use case, but we don't have an equivalent in the DT protocol.

@aslushnikov @pavelfeldman would a protocol feature make sense?  The other solutions of detecting audio in JS is unreliable.

**Comment by pavelfeldman** [comment]
Well, we have the [signal](https://cs.chromium.org/chromium/src/content/browser/web_contents/web_contents_impl.cc?type=cs&sq=package:chromium&pv=1&l=1476). The question is whether we have enough customers for the api.

**Comment by sedubois** [comment]
+1 ! I have a meditation app (web + mobile) with with videos, guided audio meditations and bell sounds during silent meditations, and want to check if the sound plays correctly. Detecting sound is actually for me the most important feature (and beyond that, I could even dream of checking what kind of sound is played).

**Comment by notedit** [comment]
+1  need this api too,  want to do some audio recorder.

**Comment by aslushnikov** [comment]
Everybody: please upvote the issue itself rather then posting "+1" comments.

**Comment by sedubois** [comment]
@aslushnikov with the release of Chromeless v1, do you know if there is any way to achieve this now?

**Comment by princejonn** [comment]
+1

**Comment by andrewpolemeni** [comment]
A great use for this would be a website virtual assistant. If headless chrome can use audio as an input you could basically implement a "Siri".

**Comment by JCMais** [comment]
This is also useful for e2e tests related to webrtc, where it could be used to detect if audio is working

**Comment by marik22312** [comment]
Great request! looking for a way to monitor an mp3 audio stream, would be very useful
